app/locations/auto.ts

```diff
@@ -0,0 +1,19 @@
+import AutoLocation from '@ember/routing/auto-location';
+
+// Sadly the `auto` location must be overriden, otherwise
+// ember-cli's server doesn't allow visiting specific URLs
+export default class GuidAutoLocation extends AutoLocation {
+    pattern = /--[^/]+\//g;
```

Comment:
Maybe make this a little more restrictive by adding a positive lookbehind to make sure it only replaces segments that *start* with --, e.g. `/(?<=^|\/)--[^/]+\//g`.